### PR TITLE
build: Add checker for PR title

### LINKED ISSUE #75: Add a Github check for PR titles
### What is the problem the feature request solves?

We should enforce PR titles to follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) style, and in the format such as "feat: XXX". 

### Describe the potential solution

Add a Github checker on the PR titles and potentially label them accordingly. Perhaps we can explore whether the marketplace already has something handy (like [this one](https://github.com/marketplace/actions/conventional-commit-in-pull-requests)?)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sunchao**
## Which issue does this PR close?

Closes #75 .

## Rationale for this change

It be useful to add a checker making sure all the PR titles conform to certain format, such as [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

## What changes are included in this PR?

Added a Github action to check PR title and fail the PR if it violates the format. At the moment, the format is simply "type: message" or "type(scope): message". We don't restrict the set of valid types or scopes yet.

### CODE REVIEW

**Review by viirya — APPROVED** [review]

File: .github/workflows/pr_build.yml [hunk 1]
```diff
@@ -126,3 +126,17 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check PR title
+        env:
+          PR_TITLE: ${{ github.event.pull_request.title }}
+        run: |
+          if ! echo $PR_TITLE | grep -Eq '^(\w+)(\(.+\))?: .+$'; then
+            echo "PR title does not follow conventional commit style."
+            echo "Please use a title in the format: type: message, or type(scope): message"
```
Comment (comphead) [inline .github/workflows/pr_build.yml:139]: I think its misleading a bit, @sunchao
maybe we can attach a title example?

Comment (sunchao) [reply .github/workflows/pr_build.yml:139]: OK, added an example

Comment (comphead) [reply .github/workflows/pr_build.yml:139]: Much nicer

**Review by comphead — APPROVED** [review]
lgtm thanks @sunchao

**Comment by sunchao** [comment]
Thanks, merged

**Pull request merged**